compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1373,8 +1375,20 @@ private boolean validateErrorVariable(BLangErrorVariable errorVariable, BErrorTy
         }
 
         if (isRestDetailBindingAvailable(errorVariable)) {
+            // Type of rest pattern is a map type where constraint type is,
+            // union of keys whose values are not matched in error binding/match pattern.
             BTypeSymbol typeSymbol = createTypeSymbol(SymTag.TYPE);
-            BMapType restType = new BMapType(TypeTags.MAP, recordType.restFieldType, typeSymbol);
+            BUnionType restFieldType = BUnionType.create(null, recordType.restFieldType);
+            for (Map.Entry<String, BField> entry : fieldMap.entrySet()) {
+                if (!matchedDetailFields.contains(entry.getKey())) {
+                    BType type = entry.getValue().getType();
+                    if (!type.equals(symTable.nilType)) {
```

Comment:
Why are we not adding if the type is nil?

Reply:
Yep we don't need this, will remove